src/lib/homestore/heap_chunk_selector.cpp

```diff
@@ -0,0 +1,131 @@
+#include "heap_chunk_selector.h"
+
+#include <execution>
+#include <algorithm>
+#include <utility>
+
+#include <sisl/logging/logging.h>
+
+namespace homeobject {
+// https://github.com/eBay/HomeObject/pull/30#discussion_r1331112743
+// we make the following assumptions
+// 1 homestore will initialize HeapChunkSelector by adding all the chunks single threaded
+// 2 we do not need dynamic chunk requirements
+
+// it means after the single thread initialization,
+// 1 the key collection of m_per_dev_heap will never change.
+// 2 the key collection of m_chunks will never change
+void HeapChunkSelector::add_chunk(csharedChunk& chunk) {
+    VChunk vchunk(chunk);
+    auto pdevID = vchunk.get_pdev_id();
+    // add this find here, since we don`t want to call make_shared in try_emplace every time.
+    auto it = m_per_dev_heap.find(pdevID);
+    if (it == m_per_dev_heap.end()) it = m_per_dev_heap.emplace(pdevID, std::make_shared< PerDevHeap >()).first;
+    auto& avalableBlkCounter = it->second->available_blk_count;
+    avalableBlkCounter.fetch_add(vchunk.available_blks());
+
+    m_chunks.emplace(vchunk.get_chunk_id(), chunk);
+
+    auto& heapLock = it->second->mtx;
+    auto& heap = it->second->m_heap;
+    std::lock_guard< std::mutex > l(heapLock);
+    heap.emplace(chunk);
+}
+
+// select_chunk will only be called in homestore when creating a shard.
+csharedChunk HeapChunkSelector::select_chunk(homestore::blk_count_t count, const homestore::blk_alloc_hints& hint) {
+    auto& chunkIdHint = hint.chunk_id_hint;
```

Comment:
Do you want to use consistent naming conventions? either myVarName or my_var_name? I see both of these in this, can we use consistent one?